From 0de0124615870b9c574360c286f46e5f9ba66b89 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Sun, 20 Nov 2022 17:45:11 +0100
Subject: [PATCH 10/39] nl80211: remove ancient wpa_supplicant ctrl socket path

This isn't required nor used anymore, and it's been +8 years.

See
"735a03be: iwinfo: adjust for changed wpa_supplicant control socket path"
on the main repository.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 iwinfo_nl80211.c | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/iwinfo_nl80211.c b/iwinfo_nl80211.c
index aceef18..c1e53d7 100644
--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -935,8 +935,7 @@ static int nl80211_wpactl_connect(const char *ifname, struct sockaddr_un *local)
 
 	remote.sun_family = AF_UNIX;
 	remote_length = sizeof(remote.sun_family) +
-		sprintf(remote.sun_path, "/var/run/wpa_supplicant-%s/%s",
-		        ifname, ifname);
+		sprintf(remote.sun_path, "/var/run/wpa_supplicant/%s", ifname);
 
 	/* Set client socket file permissions so that bind() creates the client
 	* socket with these permissions and there is no need to try to change
@@ -958,14 +957,8 @@ static int nl80211_wpactl_connect(const char *ifname, struct sockaddr_un *local)
 
 	if (connect(sock, (struct sockaddr *)&remote, remote_length))
 	{
-		remote_length = sizeof(remote.sun_family) +
-			sprintf(remote.sun_path, "/var/run/wpa_supplicant/%s", ifname);
-
-		if (connect(sock, (struct sockaddr *)&remote, remote_length))
-		{
-			close(sock);
-			return -1;
-		}
+		close(sock);
+		return -1;
 	}
 
 	local->sun_family = AF_UNIX;
-- 
2.35.1

